flang/lib/Evaluate/intrinsics.cpp

```diff
@@ -1403,6 +1403,11 @@ static const IntrinsicInterface intrinsicSubroutine[]{
             {"stat", AnyInt, Rank::scalar, Optionality::optional,
                 common::Intent::Out}},
         {}, Rank::elemental, IntrinsicClass::atomicSubroutine},
+    {"chdir",
+        {{"dir", AnyChar, Rank::anyOrAssumedRank},
```

Comment:
Why `anyOrAssumedRank`, is it allowed to pass arrays and assumed-rank to `chdir`?

`Rank::scalar` seems more appropriate, unless some existing programs use CHDIR with an array argument and relying on sequence association.